BioinformaticTools-inator/modules/dna_rna_funcs.py

```diff
@@ -0,0 +1,110 @@
+from typing import Dict, List, Union
+
+dna_dict = {
+    'A': 'T', 'G': 'C', 'T': 'A', 'C': 'G',
+    'a': 't', 'g': 'c', 't': 'a', 'c': 'g'
+    }
+rna_dict = {
+    'A': 'U', 'G': 'C', 'U': 'A', 'C': 'G',
+    'a': 'u', 'g': 'c', 'u': 'a', 'c': 'g'
+    }
+
+
+def is_valid_na(seq: str) -> None:
+    if ('u' in seq or 'U' in seq) and ('t' in seq or 'T' in seq):
+        raise ValueError("Error: simultaneous presence of 'u' and 't' "
+                         " in any register is unacceptable.")
+    if not set([symb.upper() for symb in seq]).issubset(
+            {'A', 'T', 'G', 'C', 'U'}
+            ):
+        raise ValueError(f"Error: the sequence contains "
+                         f"invalid characters: {seq}")
+
+
+def transcribe(*args: str) -> Union[List[str], str]:
+    results = []
+    for seq in args:
+        transcribed_seq = seq.replace('T', 'U').replace('t', 'u')
+        results.append(transcribed_seq)
+    if len(args) == 1:
+        return results[0]
+    return results
+
+
+def reverse(*args: str) -> Union[List[str], str]:
+    results = [seq[::-1] for seq in args]
+    if len(args) == 1:
+        return results[0]
+    return results
+
+
+def complement(*args: str) -> Union[List[str], str]:
+    results = []
+    for seq in args:
+        complement_seq = []
+        if 'u' in seq or 'U' in seq:
+            for symb in seq:
+                if symb in rna_dict:
+                    complement_seq.append(rna_dict[symb])
+                elif symb in rna_dict:
+                    complement_seq.append(rna_dict[symb])
+                else:
+                    complement_seq.append(symb)
+        else:
+            for symb in seq:
+                if symb in dna_dict:
+                    complement_seq.append(dna_dict[symb])
+                elif symb in dna_dict:
+                    complement_seq.append(dna_dict[symb])
+                else:
+                    complement_seq.append(symb)
+        results.append("".join(complement_seq))
+    if len(args) == 1:
+        return results[0]

```

Comment:
Всю это сложную конструкцию можно заменить


```suggestion
def complement(seq: str) -> str:
    if 'u' in set(seq.lower()):
        result.append("".join(rna_dict[ch] for ch in seq))
    else:
        result.append("".join(dna_rna[ch] for ch in seq))
    return result
```

Reply:
О, ну красиво!